### PR TITLE
testenv: move testenv block configuration to ansible vars

### PR DESCRIPTION
**Pull request by sodar**

### CODE REVIEW

**Comment by sodar** [comment]
recheck no bug

File: vars/demoEnvPipeline.groovy [hunk 1]
```diff
@@ -20,23 +18,23 @@ def call(playbookToRun) {
                 steps {
```
Comment (krdln) [inline vars/demoEnvPipeline.groovy]: Is it worth splitting in two stages at all, if the first stage only sets some variables?

File: ansible/group_vars/all/vars [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+provisioning:
+  name: "{{ testenv_name }}"
+  folder: "{{ testenv_vmware_folder }}"
+  portgroup_mgmt: "{{ testenv_mgmt_network }}"
+  portgroup_contrail: "{{ testenv_data_network }}"
+  vms:
+    controller:
+      template: "{{ testenv_controller_vmware_template }}"
+      nodes:
+      - name: "{{ testenv_name[:7] }}-controller"
```
Comment (krdln) [inline ansible/group_vars/all/vars]: Perhaps it'd be better to just error early if `testenv_name` is longer than 7?

**Comment by krdln** [comment]
I still don't get why the PR is called "prepend hostname with machine type", could you explain?

File: ansible/group_vars/all/vars [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Default test env templates
+testenv_testbed_vmware_template: Template-testbed-201803050718
+testenv_controller_vmware_template: Template-CentOS-7.4
+
+# Testenv block definition
+provisioning:
```
Comment (krdln) [inline ansible/group_vars/all/vars]: `provisioning` seems a little bit ambiguous, perhaps use the comment as variable name?

Comment (sodar) [reply ansible/group_vars/all/vars]: `testenv_block`?

File: vars/demoEnvPipeline.groovy (hunk 1 above)
Comment (sodar) [reply vars/demoEnvPipeline.groovy]: Not really, I will merge these stages.

File: ansible/group_vars/all/vars (hunk 2 above)
Comment (sodar) [reply ansible/group_vars/all/vars]: Trigger an error in Jenkinsfile or add some logic to `vmware-deploy-testenv.yml` playbook?

Comment (PawelMarchewkaCodiLime) [reply ansible/group_vars/all/vars]: [Refinement] @sodar I strongly agree with @krdln, job should fail if ```testenv_name ``` is longer then 7 characters.

Comment (krdln) [reply ansible/group_vars/all/vars]: If the error is in ansible but before anything side-effectful happens, I'm ok with that. I'd also check the machine name length instead of just testenv name, in case we change the naming scheme.

It'd be nice to also add a comment to param in deploy test and demo env jobs.

File: ansible/group_vars/all/vars (hunk 3 above)
Comment (krdln) [reply ansible/group_vars/all/vars]: Sgtm

**Comment by krdln** [comment]
@sodar Can you change the PR title then? :)

File: Jenkinsfile [hunk 4]
```diff
@@ -117,35 +113,47 @@ pipeline {
             steps {
                 script {
                     lock(label: 'testenv_pool', quantity: 1) {
-                        testNetwork = getLockedNetworkName()
-                        vmwareConfig = getVMwareConfig()
-                        testEnvName = getTestEnvName(testNetwork)
-                        testEnvFolder = env.VC_FOLDER
+                        // 'Cleanup' stage
+                        node(label: 'ansible') {
```
Comment (krdln) [inline Jenkinsfile]: This comment is unrelated to this block.

Also, this code could be run outside the `node` block. (if not, please explain why it needs to be run inside `node`)

File: ansible/group_vars/all/vars [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Default test env templates
+testenv_testbed_vmware_template: Template-testbed-201803050718
+testenv_controller_vmware_template: Template-CentOS-7.4
```
Comment (krdln) [inline ansible/group_vars/all/vars]: Do we need these defaults here?

**Comment by krdln** [comment]
Have you checked that deploy demo env works after this refactoring?

**Comment by sodar** [comment]
@krdln Yup:
- deploying: http://10.84.12.26:8080/job/WinContrail-infra/job/deploy-dev-env/28/
- destroying: http://10.84.12.26:8080/job/WinContrail-infra/job/destroy-env/76
- name too long: http://10.84.12.26:8080/job/WinContrail-infra/job/deploy-dev-env/30

File: Jenkinsfile (hunk 4 above)
Comment (sodar) [reply Jenkinsfile]: It needs to be in `node`, because script uses `env.WORKSPACE` which is only defined when job is running on a given node.

But yes, comment can be deleted.

File: ansible/group_vars/all/vars (hunk 5 above)
Comment (sodar) [reply ansible/group_vars/all/vars]: I was thinking about these as a substitute to Jenkins parameters/contents. I think they can stay for now and in the future we'll move jenkins variables to vars/vault.
